### PR TITLE
Add changeable dock transparency

### PR DESCRIPTION
**Pull request by ghost**
You can now enable/disable dock transparency from the settings (see pull request on cyber-settings.) or from the config file (see pull request on cyber-dock.).

image

image

### CODE REVIEW

File: src/qml/DockPage.qml [hunk 1]
```diff
@@ -112,6 +112,55 @@ ItemPage {
...
+                id: transparencyTabBar
+                Layout.fillWidth: true
+
+                TabButton {
+                    text: qsTr("Enabled")
+                }
+
+                TabButton {
+                    text: qsTr("Disabled")
+                }
```
Comment (itsvic-dev) [inline src/qml/DockPage.qml]: could you make this a switch? https://doc.qt.io/qt-5/qml-qtquick-controls2-switch.html

**Review by itsvic-dev — CHANGES_REQUESTED** [review]

File: src/qml/DockPage.qml (hunk 1 above)
Comment (ghost) [reply src/qml/DockPage.qml]: doing it rn

Comment (ghost) [reply src/qml/DockPage.qml]: @omaemae done

File: src/qml/DockPage.qml [hunk 2]
```diff
@@ -111,53 +111,30 @@ ItemPage {
...
-                        dockTransparency = false
-                        break;
-                    }
-					console.log(dockTransparency);
-                    appearance.setDockTransparency(dockTransparency)
-                }
+            RowLayout {
+            	
+            	
+
```
Comment (itsvic-dev) [inline src/qml/DockPage.qml:120]: what's with the large spaces here?

File: src/qml/DockPage.qml [hunk 3]
```diff
@@ -111,53 +111,30 @@ ItemPage {
...
+	
+            	Switch {
+                        	id: dockTransparencySwitch
+                        	Layout.fillHeight: true
+                        	checked: appearance.dockTransparency
+                        	leftPadding: 0
+                        	rightPadding: 0
+                        	onCheckedChanged: appearance.dockTransparency = checked
+            	}
+
```
Comment (itsvic-dev) [inline src/qml/DockPage.qml:135]: inconsistent tabs

File: src/qml/DockPage.qml [hunk 4]
```diff
@@ -111,53 +111,30 @@ ItemPage {
...
+            	
+            	
+            	
+            	Label {
+                	text: qsTr("Dock Transparency")
+                	color: Meui.Theme.disabledTextColor
+                	bottomPadding: Meui.Units.smallSpacing
+            	}
+	
+            	Switch {
```
Comment (itsvic-dev) [inline src/qml/DockPage.qml:127]:
```suggestion
            	}

            	Item { Layout.fillWidth: true }
            	
            	Switch {
```

File: src/qml/DockPage.qml [hunk 5]
```diff
@@ -111,53 +111,30 @@ ItemPage {
 
             Item {
                 Layout.fillHeight: true
+                Layout.fillWidth: true
```
Comment (itsvic-dev) [inline src/qml/DockPage.qml:114]: why

**Review by itsvic-dev — CHANGES_REQUESTED** [review]

File: src/qml/DockPage.qml (hunk 2 above)
Comment (ghost) [reply src/qml/DockPage.qml:120]: > what's with the large spaces here?

was trying something before, forgot to remove the spaces

File: src/qml/DockPage.qml (hunk 5 above)
Comment (ghost) [reply src/qml/DockPage.qml:114]: > why

you told me to add layout.fillwidth

File: src/qml/DockPage.qml (hunk 3 above)
Comment (ghost) [reply src/qml/DockPage.qml:135]: > inconsistent tabs

ah sorry about that

File: src/qml/DockPage.qml (hunk 5 above)
Comment (itsvic-dev) [reply src/qml/DockPage.qml:114]: in between the label and the switch, not here

Comment (ghost) [reply src/qml/DockPage.qml:114]: > in between the label and the switch, not here

oh

File: src/qml/DockPage.qml [hunk 6]
```diff
@@ -109,6 +109,32 @@ ItemPage {
                 }
             }
 
+            Item {
+                Layout.fillHeight: true
+                Layout.fillWidth: true
+            }
+            
+            RowLayout {
```
Comment (itsvic-dev) [inline src/qml/DockPage.qml:117]: another nitpick: could you add a separator here? you can see these between Position and Size

Comment (ghost) [reply src/qml/DockPage.qml:117]: doing it rn

**Pull request merged**